src/services/factories/tokenList.ts

```diff
@@ -164,62 +166,60 @@ export function getTokensFactory(factoryParams: {
     return new Map<string, number>(filteredAddressAndIds)
   }
 
-  async function updateTokenDetails(networkId: number, numTokens: number): Promise<void> {
-    // Get filtered ids and addresses
-    const filteredAddressesAndIds = await _getFilteredIdsMap(networkId, numTokens)
-
-    // Create a map of current token list addresses and tokens
-    const localAddressesMap = new Map<string, TokenDetails>(
-      tokenListApi
-        .getTokens(networkId)
-        // Remove tokens that are on the list but not registered on the contract neither on TCR
-        .filter(({ address }) => filteredAddressesAndIds.has(address))
-        // Map it with token address as key for easy access
-        .map((token) => [token.address, token]),
+  async function _fetchTokenDetails(
+    networkId: number,
+    addressToIdMap: Map<string, number>,
+    tokensConfig: TokenDetails[],
+  ): Promise<TokenDetails[]> {
+    const tokensConfigMap = new Map(tokensConfig.map((t) => [t.address, t]))
+    const tokenDetailsPromises: (Promise<TokenDetails | undefined> | TokenDetails)[] = []
+    addressToIdMap.forEach((id, tokenAddress) => {
+      // Resolve the details using the config, otherwise fetch the token
+      const token: TokenDetails | undefined | Promise<TokenDetails | undefined> = tokensConfigMap.has(tokenAddress)
+        ? tokensConfigMap.get(tokenAddress)
+        : _fetchToken(networkId, id, tokenAddress)
+
+      if (token) {
+        tokenDetailsPromises.push(token)
```

Comment:
This works, but we may get slightly unexpected order of tokens
Before it was `localTokens + fetchedTokens`, now they will be interleaved. Before new tokens that get added to TCR would appear at the end of the list, now -- who knows.

Not a big problem though.

Reply:
Should we sort alphabetically?